src/cmap/commands.ts

```diff
@@ -852,7 +852,7 @@ export class BinMsg {
         this.index += bsonSize;
       } else if (payloadType === 1) {
         // It was decided that no driver makes use of payload type 1
-        throw new MongoDriverError('OP_MSG Payload Type 1 detected unsupported protocol');
+        throw new MongoInvalidArgumentError('OP_MSG Payload Type 1 detected unsupported protocol');
```

Comment:
This is actually something a poorly behaving server would send us, its an unused payload type. So this wouldn't be an argument error, its still a driver error in the sense that the driver doesn't have the logic to handle that payload type, is there something better that fits this?

Reply:
At the moment, we don't have an error that covers this cleanly, but we could conceivably have something like a `MongoInvalidServerResponseError` or a `MongoUnexpectedServerResponseError` as a subclass of `MongoRuntimeError`

Reply:
So for the sake of this PR do we want to leave this as a driver error and we'll leave a todo with a ticket that'll bring in the new proposed error?
`MongoInvalidResponseError` seems like it could work

Reply:
I'll go ahead and change that back to a MongoDriverError and add that todo 👍🏿